src/App.tsx

```diff
@@ -1,25 +1,51 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+
+  const [perPage, setPerPage] = useState(5)
+  const [page, setPage] = useState<number>(1);
+
+  const total = 42;
+
+  const handlePerPageValueChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(Number(event.target.value));
+  }
+
+  const items = getNumbers(1, total).map(n => `Item ${n}`);
+
+  const getCurrentItems = () => {
+    const startIndex = (page - 1) * perPage;
+    const endIndex = startIndex + perPage;
+    return items.slice(startIndex, endIndex);
+  };
```

Comment:
The 'getCurrentItems' function is called every render, which can be inefficient. Consider using 'useMemo' to memoize the current items so that they are only recalculated when 'page' or 'perPage' changes.